pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/measurements/tests/Test_ExaTNCuda_Expval.cpp

```diff
@@ -0,0 +1,396 @@
+// Copyright 2024 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the License);
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an AS IS BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <algorithm>
+#include <complex>
+#include <limits>
+#include <memory>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+#include <catch2/catch.hpp>
+
+#include "ExaTNCuda.hpp"
+#include "MeasurementsTNCuda.hpp"
+#include "TNCudaGateCache.hpp"
+#include "cuda_helpers.hpp"
+
+/// @cond DEV
+namespace {
+using namespace Pennylane::LightningTensor::TNCuda::Measures;
+using namespace Pennylane::LightningTensor::TNCuda::Observables;
+} // namespace
+/// @endcond
+
+TEMPLATE_TEST_CASE("ExaTNCuda::Measurements::[Identity]", "[ExaTNCuda_Expval]",
+                   float, double) {
+    using TensorNetT = ExaTNCuda<TestType>;
+    using NamedObsT = NamedObsTNCuda<TensorNetT>;
+    auto ONE = TestType(1);
+
+    std::size_t num_qubits = 3;
+
+    TensorNetT exatn_state{num_qubits};
```

Comment:
Could we follow the same logic for gate tests?

Reply:
Yes but will take a bit of time.